hatice-ecevit-2016400138/assignment3/Dict.java

```diff
@@ -0,0 +1,84 @@
+import java.util.HashMap;
+
+public class Dict {
+    public HashMap<Character,DictNode> letters = new HashMap<Character, DictNode>();
+
+    public void insertWord (String s){
+        if (!letters.containsKey(s.charAt(0))){
+            letters.put(s.charAt(0),new DictNode());
+        }
+        DictNode current = letters.get(s.charAt(0));
+        s = s.substring(1);
+        insertWord(s,current);
+    }
+
+    /**
+     *
+     * @param s is the word that is inserted.
+     * @param node is the current node in the dictionary tree.
+     */
+    private void insertWord (String s, DictNode node) {
+        if (node.getChildren().containsKey(s.charAt(0))) {

```

Comment:
Optional:
It is often better to do 
X = foo.get(key);
if (X != null) {
  ...
}
instead of
if (foo.containsKey(key)) {
   X = foo.get(key);
}

This is for a few reasons:
 1) Performance. The first requires 1 lookup, the second requires 2.
 2) Correctness. In multithreaded situations, the map could have changed between the call to containsKey and the lookup. With the first, you know your result is consistent, as you only do one lookup.
But it is only a nit, and the performance increase isn't a big O change, so it isn't that important.